ls1intum/Artemis
# `General`: Fix an issue in the course overview and an edge case in programming exercise solution participations

### Checklist
#### General


#### Server

### Motivation and Context


Fix latest sentry issues:
* [Course overview](https://sentry.ase.in.tum.de/organizations/artemis/issues/31675/?environment=prod&query=is%3Aunresolved&referrer=issue-stream)
    * Also caused the [unknown translations](https://sentry.ase.in.tum.de/organizations/artemis/issues/31560/?environment=prod&query=is%3Aunresolved&referrer=issue-stream)
* [Exercise issue](https://sentry.ase.in.tum.de/organizations/artemis/issues/31591/?environment=prod&query=is%3Aunresolved&referrer=issue-stream)
    * Also reported here: Fixes #6556

### Description

Quiz exercises can have a submission without a participation due to their scheduling. This is handled in the code, where it tried to add the new participation to an unchangeable list.
The other issue was that when a new result for the solution was produced it was possible that the default branch is not yet stored in Artemis, causing the null pointer.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 2 Students
- 1 Programming Exercise
- 1 Quiz Exercise

1. Log in to Artemis
2. Create the exercise and commit a second time to its solution, the new result should be shown
3. Participate as a student in the quiz and reload the exercise overview during the conduction of the quiz. Check that there are no errors and you correctly see your running participation.

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage



| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|-----------------------------:|
| ExerciseService.java | 97% | ✅                           |
| AbstractVersionControlService.java | 95% | ✅              |
